clang/docs/checkedc/Update-to-latest-LLVM-sources.md

```diff
@@ -1,81 +1,212 @@
+   release of LLVM/Clang:
+      - We first upgrade the `release_12.x` branch (of the `checkedc-clang`
+      repository) to the sources corresponding to the 12.0.0 release on the
+      `release/12.x` branch of the LLVM/Clang repository.
+      - Next, we merge Checked C features and bug fixes from the `master` branch
+      to the `release_12.x` branch.
+      - The release of the Checked C compiler happens from the `release_12.x`
+      branch.
+   - When we wish to make a Checked C compiler release based on the 12.0.1
+   release of LLVM/Clang:
+      - We upgrade the same `release_12.x` branch (of the `checkedc-clang`
+      repository) to the sources corresponding to the 12.0.1 release on the
+      `release/12.x` branch of the LLVM/Clang repository.
+      - Next, we merge Checked C features and bug fixes from the `master` branch
+      to the `release_12.x` branch.
+      - The release of the Checked C compiler happens from the `release_12.x`
+      branch.
+      
+
+We do not expect substantial developer effort to be expended in Phase 2 as it is
+highly unlikely that the bug fixes on the release branches of the LLVM/Clang
+repository will break Checked C build and tests. 
+
+Note that Phase 1 and the sub-phases of Phase 2 may not be executed one after
+another in one continuous stretch. They may be spread over several weeks and may
+be interspersed with Checked C development on the `master` branch.
+
+## Detailed Instructions - Phase 1
+
+1. Create a new branch off the `baseline` branch in the `checkedc-clang`
+repository. Let us call this new branch `updated_baseline`.
+```        
+    git checkout -b updated_baseline baseline
+```     
+2. Upgrade the branch `updated_baseline` to &lt;branch-point-of-12-on-main&gt;
+commit hash on the branch `main` of the LLVM/Clang repository. Record the commit
+hash &lt;branch-point-of-12-on-main&gt; in the document LLVM-Upgrade-Notes.md.
+```
+    git remote add upstream https://github.com/llvm/llvm-project.git 
+    git pull upstream <branch-point-of-12-on-main>
+```
+3. Execute  LLVM/Clang tests on the branch `updated_baseline` on your local
+machine to make sure that the upgrade is stable.
+
+4. Merge the `updated_baseline` branch back to the `baseline` branch.
+```
     git checkout baseline
-    git checkout -b updated-baseline
-    git pull mirror master
-
-## Run testing on your branched baseline branches.
-
-You can run testing locally or push your branched baseline branches to GitHub
-and use automated testing.  This will show you whether there are any unexpected
-failures in your baseline branch.
-
-If you use automated testing, make sure to clean the build directory first.
-Enough changes will likely have accumulated that things may go wrong without doing
-that first.
-
-## Branch your new baseline branches and merge master changes
-
-You can now branch your baseline branches to create a new master branch:
-
-    git checkout -b updated-master
+    git merge updated_baseline
```

Comment:
```suggestion
    git merge --ff-only updated_baseline
```
We expect `updated_baseline` to be ahead of `baseline`.  If it's not, we should flag the problem for investigation rather than generate a merge commit.